crates/rpc/rpc-types/src/eth/transaction/request.rs

```diff
@@ -111,7 +110,156 @@ impl TransactionRequest {
                     access_list: access_list.unwrap_or_default(),
                 }))
             }
+            #[allow(unreachable_code)]
+            #[allow(unreachable_patterns)]
+            // EIP4844
+            (None, Some(_), access_list) | (None, None, access_list @ None) => {
+                Some(TypedTransactionRequest::EIP4844(crate::Eip4844TransactionRequest {
+                    chain_id: 0,
+                    nonce: nonce.unwrap_or_default(),
+                    max_priority_fee_per_gas: max_priority_fee_per_gas.unwrap_or_default(),
+                    max_fee_per_gas: max_fee_per_gas.unwrap_or_default(),
+                    gas_limit: gas.unwrap_or_default(),
+                    kind: match to {
+                        Some(to) => TransactionKind::Call(to),
+                        None => TransactionKind::Create,
+                    },
+                    value: value.unwrap_or_default(),
+                    gas_price: gas_price.unwrap_or_default(),
+                    access_list: access_list.unwrap_or_default(),
+                    input: data.unwrap_or_default(),
+                    blob_versioned_hashes: todo!(),
+                    max_fee_per_blob_gas: todo!(),
+                    sidecar: todo!(),
+                }))
+            }
```

Comment:
this is currently not possible and needs some additional changes,

please remove for now and open an issue for blob support in transactionaction request, which we can add separately.